### PR TITLE
[MIG] Migrated sale_order_line_price_history to v13.

### PR DESCRIPTION
**Pull request by sudhir-erpharbor**

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
You can try to convert the onchange in the transient to new computed readonly=False field.

**Comment by pedrobaeza** [comment]
@ernestotejeda please review this TT22594

**Review by ernestotejeda — COMMENTED** [review]
@sudhir-erpharbor  would you see why the runbot is red? please

**Comment by sudhir-erpharbor** [comment]
> @sudhir-erpharbor would you see why the runbot is red? please

The runbot has issue with the python version. I found following details in the runbot logs.

>File "/home/odoo/OCB-13.0/odoo/__init__.py", line 19, in <module>
assert sys.version_info > (3, 6), "Outdated python version detected, Odoo requires Python >= 3.6 to run."
AssertionError: Outdated python version detected, Odoo requires Python >= 3.6 to run.

File: sale_order_line_price_history/wizards/sale_order_line_price_history.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                                        string="Set price"
+                                        type="object"
+                                        name="action_set_price"
+                                        icon="fa-check"/>
+                            </div>
+                            <group>
+                                <field name="sale_order_line_id" invisible="1"/>
+                                <field name="history_sale_order_line_id" invisible="1"/>
+                                <field name="order_id"/>
+                                <field name="partner_id"/>
```
Comment (ernestotejeda) [inline sale_order_line_price_history/wizards/sale_order_line_price_history.xml:30]:
```suggestion
                                <field name="partner_id" attrs="{'column_invisible': [('parent.partner_id', '=', False)]}"/>
```
With this we can remove the first statement in ROADMAP section:
https://github.com/OCA/sale-workflow/pull/1082/files#diff-93b55544ac6a86454091405656f9ca6dR55

**Review by ernestotejeda — CHANGES_REQUESTED** [review]
I can't see pre-commit metafiles, maybe that's why the runbot is red, perhaps you sould rebase on 13.0 branch

**Comment by pedrobaeza** [comment]
Please include #1215

**Comment by dennybiasiolli** [comment]
I was re-creating the migration for this module, but then I discovered this PR. I would be happy to contribute in this migration, @sudhir-erpharbor do you need help?

**Comment by sudhir-erpharbor** [comment]
> I was re-creating the migration for this module, but then I discovered this PR. I would be happy to contribute in this migration, @sudhir-erpharbor do you need help?

Sorry, I won't be able to finish it in time. So if you wish, you can take over the PR or improve it.

File: sale_order_line_price_history/wizards/sale_order_line_price_history.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                                <field name="sale_order_date_order"/>
+                                <field name="product_uom_qty"/>
+                                <field name="price_unit"/>
```
Comment (dennybiasiolli) [inline sale_order_line_price_history/wizards/sale_order_line_price_history.xml:33]: after this line you should add

```
<label for="discount" groups="sale.group_discount_per_so_line"/>
<div name="discount" groups="sale.group_discount_per_so_line">
    <field name="discount" class="oe_inline"/> %%
</div>
```

File: sale_order_line_price_history/wizards/sale_order_line_price_history.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                            </group>
+                        </form>
+                        <tree>
+                            <field name="order_id"/>
+                            <field name="partner_id"/>
+                            <field name="sale_order_date_order"/>
+                            <field name="product_uom_qty"/>
+                            <field name="price_unit"/>
```
Comment (dennybiasiolli) [inline sale_order_line_price_history/wizards/sale_order_line_price_history.xml:41]: after this line you should add
```
<field name="discount" groups="sale.group_discount_per_so_line"/>
```

File: sale_order_line_price_history/wizards/sale_order_line_price_history.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                                <field name="order_id"/>
+                                <field name="partner_id"/>
+                                <field name="sale_order_date_order"/>
+                                <field name="product_uom_qty"/>
+                                <field name="price_unit"/>
+                            </group>
+                        </form>
+                        <tree>
+                            <field name="order_id"/>
+                            <field name="partner_id"/>
```
Comment (dennybiasiolli) [inline sale_order_line_price_history/wizards/sale_order_line_price_history.xml:38]: this line, in 12.0 was
```
<field name="partner_id"
       column_invisible': [('parent.partner_id', '!=', False)]}"/>
```

@ernestotejeda how can we do in this case?

**Review by dennybiasiolli — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
@sudhir-erpharbor So, we can take into account #1289 instead and close this?

**Pull request closed without merging**